java/fundamentals/tips-ii/how-range-operations-work.md

```diff
@@ -1,46 +1,55 @@
 ---
 author: catalin
 
-levels:
-
-  - basic
-
 type: normal
 
 category: must-know
 
 aspects:
-
   - introduction
 
 notes: 'https://insights.enki.com/#/contrib/others/55c83ba83f06ab320051aa8a?search=kha'
 
 ---
 
-# How range operations work
+# How Range Operations Work
 
 ---
 ## Content
 
-When specifying a range of values, the first value `from` is included while the last value `to` is excluded.
+You can use the `substring()` method to extract a part of a string, **as a new string**.
+
+The syntax is:
 
 ```java
-String.substring(from,to); 
+String substring(
+  int startIndex,
+  int endIndex
+);
 ```
-For example when applying the method listed above on the following string 
+Let's take a look at an example:
+
 ```java
-"Enki".substring(0,2);
+"Enki".substring(0, 2);
 ```
-the result will be `"En"`, as indexing begins from the `0` value. `"E"` was included ( `0` index) while `"k"` was excluded ( `2` index).
 
-There are many more methods needing specifying a range, all of them using this rule.
+The result will be `"En"`. This is because the method starts at the value of `startIndex`, includes everything in-between, but the value of `endIndex` **is excluded**.
+
+There are many more methods that need specifying a range, and all of them use this rule.
 
 ```java
-List.sublist(from,to);
+List.sublist(
+  int startIndex,
+  int endIndex
+);
 ```
 
 ```java
-Array.copyOfRange(T[],from,to);
+Array.copyOfRange(
```

Comment:
this is very confusing I have no idea what it refers to 🤣

instead, we can remove it and show a concrete example of lists. define a list with 4 values and retrieve one of its sublists

Reply:
something like this:

```java
List<Integer> myList = new ArrayList<Integer>();
for (int i = 0; i <= 4; i++) {
  myList.add(i);
}
List<Integer> newList = myList.subList(0, 3);
System.out.println(newList);
```

or something like this

```java
myList = [0, 1, ,2, 3, 4]
mySublist = myList.subList(0, 3);
// [0, 1, 2]
```

Reply:
I think the first example yeah - it points out how `.sublist` works based on ranges